api/v1alpha1/oauth2client_types.go

```diff
@@ -129,7 +129,7 @@ type OAuth2ClientSpec struct {
 // GrantType represents an OAuth 2.0 grant type
 type GrantType string
 
-// +kubebuilder:validation:Enum=id_token;code;token
+// +kubebuilder:validation:Enum=id_token;code;token;code token;token code;code id_token;id_token code;id_token token;token id_token;code id_token token;id_token token code;token code id_token;code token id_token
```

Comment:
hmm, I don't think it's a good idea to put enum validation here. We should validate type itself (array of strings). 
In general hydra is responsible for validating if provided response type is valid for given fow.  Here extending enum is in my opinion bit pointless a) it's cumbersome to extend it with all possible combinations just because user want's to use "token id_token" b) some combintation provided here does not make sense from oauth2 pov

Reply:
If we want to keep to CRD validation, and not include internal validation mechanism, then we are bound by [kubebuilder](https://book.kubebuilder.io/reference/markers/crd-validation.html). Dunno what would better then the enum here, a regexp pattern?

Reply:
We agreed to stay for now with enum bit cleaned up and following https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html